src/editor/Editor.jsx

```diff
@@ -2,53 +2,251 @@ var React = require('react');
+                        	onChange={this._handleMenuClick.bind(this, "formatblock")} />
+                        <DropDownIcon 
+	                    	iconClassName="cfi cfi-text-height" 
+	                    	menuItems={fontSizeOptions}
+	                    	onChange={this._handleMenuClick.bind(this, "fontsize")} />
+                        <DropDownIcon 
+                        	iconClassName="cfi cfi-font" 
+                        	menuItems={fontNameOptions}
+                    		onChange={this._handleMenuClick.bind(this, "fontname")} />
+                </ToolbarGroup>
                 </Toolbar>
                 <div>
-                    <TextFieldRich />
+                    {displayEditor}
                 </div>
+                <Dialog ref="linkDialog" title="Enter the link path" actions={dialogActions} modal={true} >
+                	<TextField ref="linkInput" />
+                </Dialog>
             </div>
         )
     },
 
+    /**
+     * Callback used to handle commands when button is clicked on the toolbar
+     *
+     * @param {string} type		The name of the command to execute
+     * @private
+     */
+    _handleToolbarClick: function(type) {
+    	if (!this.isMounted()) { 
+    		return;
+    	}
+    	
+    	// If current display is source view, we dont need to continue unless toggle src is clicked
+    	if(this.state.sourceViewMode && type != "src") {
+    		return;
+    	}	
+    	
+    	switch(type) {
+    		case "src":
+    			this._toggleEditorView();
+    			break;
+    		case "link":
+    			this.refs.linkDialog.show();
+    			break;
+    		case "table":
+    			this.refs.rte.insertHtml('<table style="border: 1px solid; padding: 10px;"><tbody><tr><td>&nbsp;</td><td>&nbsp;</td></tr></tbody></table>');
+    			break;
+    		case "backcolor":
+    			this.refs["forecolorPicker"].close();
+    		case "forecolor":
+    			// Hide the currently opened color pickers
+    			this.refs["backcolorPicker"].close();
+    			
+    			this.refs[type + "Picker"].show();
+    			this._colorFontType = type;
+    			break;
+    		default: 
+    			this.refs.rte.sendCommand(type, '');
+    			break;
+    	}
+    },
+    
+    /**
+     * Callback used to handle commands when button is clicked on the toolbar
+     *
+     * @param {string} type		The name of the command that triggered the menu click
+     * @param {DOMEvent} e 		Reference to the DOM event being sent
+     * @param {Integer} key		The index of the menu clicked
+     * @param {Object} payload	The object value of the menu clicked
+     * @private
+     */
+    _handleMenuClick: function(type, e, key, payload) {
+    	this.refs.rte.sendCommand(type, payload.payload);
+    },
+    
+    /**
+     * Accepts the link from the dialog box and sends a command to create the a href link
+     *
+     * @private
+     */
+    _handleDialogSubmit: function() {
+    	var input = this.refs.linkInput.getValue();
+    	
+    	this.refs.rte.insertLink(input);
```

Comment:
Check to make sure this.refs.rte is not null before using here since it is not guaranteed.